src/App.jsx

```diff
@@ -272,12 +290,21 @@ function App() {
     }
     updatedBoard[rowIndex][colIndex] = updatedTile;
 
+    let updatedFlagLocations = [];
+
     if ( isFlagged ){
-      setRemainingFlagsCount(remainingFlagsCount - 1)
+      flagCount  = flagCount - 1;
+      updatedFlagLocations = [...flagLocations, {x: selectedTile.x, y: selectedTile.y}];
     }
     else {
-      setRemainingFlagsCount(remainingFlagsCount + 1);
+      flagCount = flagCount + 1;
+      updatedFlagLocations = flagLocations.filter(
+        (flagLocation) =>
+          flagLocation.x !== selectedTile.x && flagLocation.y !== selectedTile.y
+      );
     }
+    setRemainingFlagsCount(flagCount);
+    setFlagLocations(updatedFlagLocations);
```

Comment:
Another good practice is to avoid `let` declarations; in most cases, try to use `const`.

In this case, this would force you to create a new function to abstract the logic of that `if` condition, which will make the code more readable.

```js
const { updatedFlagLocations, remainingFlagCount } = updateFlagLocations(flagCount, flagLocations);

setRemainingFlagsCount(remainingFlagCount);
setFlagLocations(updatedFlagLocations)
```

Reply:
To be clear, `let` declarations can still be used, but it's better to avoid them when it's possible